portfolio/src/main/webapp/script.js

```diff
@@ -43,14 +43,59 @@ function deleteData() {
     });
 }
 
+// Create the script tag, set the appropriate attributes
+var script = document.createElement('script');
+script.src = 'https://maps.googleapis.com/maps/api/js?key=AIzaSyDjhk-U7NK6nQwmqSScCuZvO3l3_Q88Kps';
+script.defer = true;
+script.async = true;
+
+// Attach your callback function to the `window` object
+window.initMap = function() {
+  // JS API is loaded and available
+};
+
+// Append the 'script' element to 'head'
+document.head.appendChild(script);
+
 function createMap() {
+    document.head.appendChild(script);
   const map = new google.maps.Map(
   document.getElementById('map'),
-  {center: {lat: 37.422, lng: -122.084}, zoom: 16});
-}
+  {center: {lat: 35.2271, lng: -80.8431}, zoom: 10});
+
+const trexMarker = new google.maps.Marker({
```

Comment:
these seem like they can be merged into a list. Then, the addListeners can be done via a loop, which helps reduce repetition.

Reply:
acknowledged

Reply:
finally solved.